### ISSUE TITLE
Feature: make configuration.yaml first rate citizen again

### ISSUE THREAD

**Issue by TeraHz** [Olen/homeassistant-plant#69]
TL; DR; I'd like to put in a request to reconsider dropping configuration.yaml support. It is extremely useful when working with many plants.

Long version:
I have dozens of plants with their own BT sensors (the cheap grow veg ones that sell in bulk for 2-3 bucks on eBay) that are monitored by an ESPhome board. To integrate a plant fully I have to:
- Edit the esphome config to add the the many lines of code for receiving and battery probing
- Create the plant device via the UI and select the right sensors
  - Edit the plant to adjust illumination/dli status 
- Create the plant card and add battery sensor

This is surprisingly long process with many many many clicks and searches through hundreds of entities.

I've written a yaml generator that creates all the yaml for the above steps from a single CSV file with BT MAC,plant name entries. As long as I can write it out to configuration.yaml (and esp yaml and dashboard yaml) everything works. 

If HA allowed for the creation of these devices via REST, I'd be fine, but it doesn't look like the Rest API has that exposed.

Please reconsider abandoning configuration.yaml

**Comment by Olen** [comment]
Hi, and thanks for the feedback.
I don't think I will make yaml-config a "first rate citizen" again, as HA is clearly moving away from yaml-config for devices.  Adding devices with yaml also requires a restart of HA which is not ideal.

HOWEVER - I would be happy to accept a PR that for instance will run the conversion from yaml to GUI-config each time HA is restarted.  That way you can add new devices to the yaml-config, while still managing them in the GUI after they have been coverted. Such an implementation would need to make sure devices are only added once.  It would also need to update the yaml schema to allow for all the new options and changes to how a plant is working in the "new" integration. 

Another option which I much prefer is the suggestion of includng a REST-API to add a plant.  But again, someone needs to create a PR as I don't have the capacity to do that for now, and I don't really need it personally.  I have a lot of plants, but they were added once, and I don't add new ones often enough to make it worth it.

But please create a PR and I'll look into it.

**Comment by Olen** [comment]
Cleaning up a few old issues
